### ISSUE TITLE
connection setPrefetchRowCount not working as of 1.6.0

### ISSUE THREAD

**Issue by designsinnovate** [strongloop/strong-oracle#30]
Hi,

Up until version 1.5.1, connection.setPrefetchRowCount worked as expected. Getting a 1000 rows with prefetch of 500 took less than a second.

From v1.6.0 onwards, it seems like this setter has no effect. Getting the same 1000 rows now takes over a minute - the same time as it was taking in 1.5.1 without setting the prefetch row count.

Please have a look - I noticed quite a few changes in connection.cpp revolving around prefetch logic.

Thanks.

**Comment by raymondfeng** [comment]
Please try 1.6.2 release.

Thanks,

---
Raymond Feng
Co-Founder and Architect @ StrongLoop, Inc.

StrongLoop <http://strongloop.com/> makes it easy to develop APIs <http://strongloop.com/mobile-application-development/loopback/> in Node, plus get DevOps capabilities <http://strongloop.com/node-js-performance/strongops/> like monitoring, debugging and clustering.

**Comment by designsinnovate** [comment]
The following additional change fixed the issue in 1.6.2

After some digging into the code, found this line commented out in reader.cpp
```
// baton->count = baton->connection->getPrefetchRowCount();
```
Replacing with following line fixed the issue.
```
baton->count = baton->m_prefetchRowCount;
```
baton->count is being used in EIO_NextRows method.

Hope that helps.

**Comment by raymondfeng** [comment]
@designsinnovate How could I miss that? Anyway, it's fixed in 1.6.3.
